### PR TITLE
Review and update developer documentations #6228

### LINKED ISSUE #6228: Review and update developer documentations
The time is ripe to re-organize and update many parts of our developer-facing documentations.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #6228 

Most of the work here involve updating (many parts are very outdated), simplifying, and clearing/cleaning up. The extended commit message contains the rationale for each commit.

The changes can be viewed [on the branch itself](https://github.com/TEAMMATES/teammates/tree/6228-update-docs). Documents of particular interest:
- `.gitattributes`
- `README.md`, the very front page
- `docs/README.md`, the sitemap for documentation (newly added)
- `docs/settingUp.md`
- `docs/development.md` (newly added)
- `docs/orientation-guide.md`
- `docs/troubleshooting-guide.md`

~~Still pending (also may be broken up to two or more files): `docs/process.md`.~~ Reverted again.

~~- `docs/process.md`~~
~~- `docs/team.md` (newly added)~~

### CODE REVIEW

File: docs/settingUp.md [hunk 1]
```diff
@@ -1,333 +1,86 @@
...
-   More info on this can be found [here](godmode.md). Sometimes, Eclipse does not show
-   these options immediately after you set up the project. 'Refreshing' the project should fix that. 
+More information can be found at [this documentation](https://help.github.com/articles/fork-a-repo/).
 
-   When running the test cases, if a few cases fail (this can happen due to timing issues),
-   run the failed cases using the `Run Failed Test` icon in the TestNG tab in Eclipse
-   until they pass. 
+## Step 3: Set up project-specific settings and dependencies
 
+1. Run this command to download the correct version Google App Engine SDK as used in the project:
```
Comment (damithc) [inline docs/settingUp.md]: Some folks couldn't figure out where to run this command. Can mention it here.

File: docs/settingUp.md [hunk 2]
```diff
@@ -1,333 +1,86 @@
...
+      This value must be a valid **JDK 1.7** directory.
+   * `src/test/resources/test.properties`<br>
+      Append a **same** unique id (e.g your name) to **each** of the default accounts found at the bottom of this file,
+      e.g change `test.student1.account=alice.tmms` to `test.student1.account=alice.tmms.KevinChan`.
 
+## Step 4: Set up Eclipse IDE
 
-## Troubleshooting
-Troubleshooting instructions are given [in this document](troubleshooting-guide.md)
+Eclipse IDE is our preferred development environment and support request for TEAMMATES-specific Eclipse issues will be entertained.
```
Comment (damithc) [inline docs/settingUp.md]: ~~and support request for TEAMMATES-specific Eclipse issues will be entertained~~

No added value in stating this.

File: docs/settingUp.md [hunk 3]
```diff
@@ -1,333 +1,86 @@
...
 
-## Tool stack
+> Change all references of `Eclipse → Preferences → ...` to `Window → Preferences → ...` in Windows.
```
Comment (damithc) [inline docs/settingUp.md]: >In the instructions below,  replace `Eclipse → Preferences → ...` with`Window → Preferences → ...` if you are using Windows.

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+Here are some documents important for TEAMMATES developers.
+
+## Main documents
+
+* [**Specification**](https://docs.google.com/document/d/1hjQQHYM3YId0EUSrGnJWG5AeFpDD_G7xg_d--7jg3vU/pub?embedded=true) (short name: **Spec**): Product specification, written from the developers point of view. The user's point of view is [here](http://teammatesv4.appspot.com/features.jsp).
+* The project [**vision, challenges, and principles**](overview.md) from a developer's perspective.
+* Software [**Design**](design.md) - Architecture, APIs, etc.
+* How to [**set up the developer environment**](settingUp.md).
+* The individual development [**process**](development.md) - how to work with your own local copy of TEAMMATES.
+* The team development [**workflow**](process.md).
```
Comment (damithc) [inline docs/README.md]: * The [**workflow/process**](process.md) to follow when contributing to TEAMMATES.
* [**Coding/testing instructions**](development.md) for TEAMMATES developers.

File: docs/development.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
+# Development
```
Comment (damithc) [inline docs/development.md]: You can give an overview of the document at the top before listing the ToC as the reader should know what this document is about (this applies to other documents too). In that overview, you can also mention what `dev server` means.

File: docs/development.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+* [Managing the dev server](#managing-the-dev-server)
+* [Logging in to the system](#logging-in-to-the-system)
```
Comment (damithc) [inline docs/development.md]: `Logging in to a TEAMMATES instance`?

File: docs/settingUp.md (hunk 1 above)
Comment (nikhilmudholkar) [inline docs/settingUp.md]: Yup this would be helpful. Even it took time for me to figure it out where to run it, on git bash or cmd.

Comment (wkurniawan07) [inline docs/settingUp.md]: Top of this document:

> - All the commands are assumed to be run from the root project folder, unless otherwise specified.

File: docs/README.md (hunk 4 above)
Comment (wkurniawan07) [inline docs/README.md]: Coding/testing instructions might have a name clash with [this document](https://docs.google.com/document/d/14EFJzdhp10qQ9iZ-FwsS1FAf42voe4exrlbefo_zYaU/pub?embedded=true) and [this document](https://docs.google.com/document/d/1aK-1ubIA59fbNsEujwvWKqx7itwLPTD6mzFo4F_oEoI/pub?embedded=true).

Comment (damithc) [inline docs/README.md]: `Development instructions`?

File: docs/development.md (hunk 5 above)
Comment (wkurniawan07) [inline docs/development.md]: `Dev server`, `Staging server`, `Client scripts` are all explained in their respective sections.

Comment (damithc) [inline docs/development.md]: ToC already contains the term 'dev server'. A term should be be used before it is defined.

Comment (damithc) [inline docs/development.md]: Perhaps defining the terms at the beginning is not so important as sometimes the ToC can contain undefined terms. However, it is still good to give an overview of the document at the beginning because now that we are fragmenting our docs to many different files, it's good to define the purpose of each document clearly.

File: docs/settingUp.md (hunk 2 above)
Comment (wkurniawan07) [inline docs/settingUp.md]: Actually my intention here was that support request for non-Eclipse issues will not be entertained. I feel like it needs to be surfaced because our current workflow is very much IDE-independent (Travis CI _is_ IDE-independent) and using IntelliJ (I've seen at least 3 contributors trying to set up with IntelliJ) or Netbeans or other IDEs _might_ be possible, it's just that we only provide support for Eclipse.
Should I rephrase it to something like "...environment. We do not provide support for other IDEs."?

Comment (damithc) [inline docs/settingUp.md]: Yes, best to say it directly. `Support requests related to other IDEs will not be entertained`.

**Comment by wkurniawan07** [comment]
@damithc requested changes made. In addition, `process.md` is now cleaned up and so here are the additions to the documents of particular interest:

- `docs/process.md`
- `docs/team.md` (newly added)

File: docs/team.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+* Josephine Kwa ([@JosephineKwa](https://github.com/JosephineKwa))
+* Wilson Kurniawan ([@wkurniawan07](https://github.com/wkurniawan07))
+* Yap Jun Hao ([@junhaoyap](https://github.com/junhaoyap))
+
+Area leads are the team members in charge of particular aspects of the project.
+
+Area leads have the following duties:
+* Act as the first point of contact for an issue requiring triaging or a PR requiring review,
+  if the issue falls in their portfolio (as determined by the issue labels).
+* Perform maintenance tasks regularly in their portfolio area. For example:
```
Comment (wkurniawan07) [inline docs/team.md]: I intend to make a new documentation for project maintainers (area leads and above?) which will document these tasks (and the ones in project lead's duties) in greater detail, but for now organizing it this way should suffice.

That document will also contain things such as how to choose reviewers (the `m-*` label, whose mentions I have temporarily removed in this PR) which regular contributors and committers need not be involved.

**Comment by damithc** [comment]
@wkurniawan07 I'm in the middle of extracting common parts of our project processes into https://github.com/oss-generic/process/tree/restructure-content
Should be done in another day or so.
Once that is reasonably complete, individual projects are expected to refer to that and document deviations only.  May be we put this on hold until then?

**Comment by wkurniawan07** [comment]
@damithc Roger that. In fact I could make a PR there, the re-organizing that I've done here looks very similar 😮 (no I didn't know the existence of that branch before making the changes here).

**Comment by wkurniawan07** [comment]
@damithc I reverted all changes related to `process.md`; sync-ing that document and other relevant ones with the ones in `oss-generic` deserves its own issue. The others are ready for review now.

File: docs/development.md [hunk 8]
```diff
@@ -0,0 +1,225 @@
+# Development Guideline
```
Comment (damithc) [inline docs/development.md]: Guidelines?
